### PR TITLE
Project Ironside: WinLogs Page - Sort logs based on timestamp and enable text filtering

### LINKED ISSUE #3073: WinLogs Page: Allow users to filter out data in WinLogs
### Suggested new feature or improvement

If we add additional ETW Providers, the chances of high-volume Informational entries goes up. For example, if we add the Microsoft-Windows-XAML or Microsoft-Windows-XAML-Diagnostics providers, we'll get a gazillion Information category entries. We probably should add those (and several other useful) providers. So, we should provide a way for the user to set the verbosity level. Eg Errors only, or Warnings & Errors, or All.

### Scenario

This should be a dynamic filter. That is, we should still have all the items available in our lists, so that if the user decides she needs to change the verbosity level on the fly, we can accommodate that and start showing/hiding the relevant items.

### Additional details

_No response_

### PR DESCRIPTION
**Pull request by zadesai**
## Summary of the pull request
Use AdvancedCollectionView to sort and filter logs

## Validation steps performed
1. Launch PI
2. Attach to ProblemApp2
3. Go to WinLogs page
4. Check logs are in ascending order
5. Enter text in the text box to filter logs

## PR checklist
- [ ] Closes #3073

### CODE REVIEW

File: tools/PI/DevHome.PI/ViewModels/WinLogsPageViewModel.cs [hunk 1]
```diff
@@ -65,9 +72,13 @@ public WinLogsPageViewModel()
 
         _insightsService = Application.Current.GetService<PIInsightsService>();
 
+        _filterMessageText = string.Empty;
         _winLogEntries = [];
         _winLogsOutput = [];
         _winLogsOutput.CollectionChanged += WinLogsOutput_CollectionChanged;
+        _winLogsView = new AdvancedCollectionView(_winLogEntries, true);
```
Comment (timkur) [inline tools/PI/DevHome.PI/ViewModels/WinLogsPageViewModel.cs:79]: >AdvancedCollectionView [](http://example.com/codeflow?start=27&length=22)

Looks like I should look into this for the WER page as well... looks like it does all the heavy lifting!

**Review by timkur — APPROVED** [review]

File: tools/PI/DevHome.PI/ViewModels/WinLogsPageViewModel.cs (hunk 1 above)
Comment (zadesai) [reply tools/PI/DevHome.PI/ViewModels/WinLogsPageViewModel.cs:79]: Yes, it is a great option. However, was difficult to find it.

File: tools/PI/DevHome.PI/Pages/WinLogsPage.xaml [hunk 2]
```diff
@@ -49,8 +50,9 @@
                 <TextBlock Text="&#xE74D;" FontFamily="{StaticResource SymbolThemeFontFamily}" FontSize="{StaticResource SubtitleTextBlockFontSize}"/>
             </Button>
         </StackPanel>
+        <TextBox Grid.Row="3" Text="{x:Bind ViewModel.FilterMessageText, Mode=TwoWay, UpdateSourceTrigger=PropertyChanged}" TextChanged="{x:Bind ViewModel.UpdateWinLogsView}"/>
```
Comment (AmelBawa-msft) [inline tools/PI/DevHome.PI/Pages/WinLogsPage.xaml:53]: > The [RelayCommand](https://learn.microsoft.com/en-us/dotnet/api/microsoft.toolkit.mvvm.input.RelayCommand) and [RelayCommand<T>](https://learn.microsoft.com/en-us/dotnet/api/microsoft.toolkit.mvvm.input.RelayCommand-1) are ICommand implementations that can expose a method or delegate to the view. These types act as a way to bind commands between the viewmodel and UI elements.

Reference:
- https://learn.microsoft.com/en-us/dotnet/communitytoolkit/mvvm/relaycommand

```suggestion
        <TextBox Grid.Row="3" Text="{x:Bind ViewModel.FilterMessageText, Mode=TwoWay, UpdateSourceTrigger=PropertyChanged}" >
            <i:Interaction.Behaviors>
                <ic:EventTriggerBehavior EventName="TextChanged">
                    <ic:InvokeCommandAction Command="{x:Bind ViewModel.UpdateWinLogsCommand}" />
                </ic:EventTriggerBehavior>
            </i:Interaction.Behaviors>
        </TextBox>
```

File: tools/PI/DevHome.PI/ViewModels/WinLogsPageViewModel.cs [hunk 3]
```diff
@@ -110,6 +121,11 @@ public void UpdateTargetProcess(Process process)
         }
     }
 
+    public void UpdateWinLogsView()
```
Comment (AmelBawa-msft) [inline tools/PI/DevHome.PI/ViewModels/WinLogsPageViewModel.cs:124]:
```suggestion
    [RelayCommand]
    public void UpdateWinLogs()
```

**Review by AmelBawa-msft — APPROVED** [review]

**Pull request merged**